mailbox/postgres/src/test/java/org/apache/james/mailbox/postgres/user/PostgresSubscriptionMapperRowLevelSecurityTest.java

```diff
@@ -0,0 +1,86 @@
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.postgres.user;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.james.backends.postgres.PostgresExtension;
+import org.apache.james.backends.postgres.utils.JamesPostgresConnectionFactory;
+import org.apache.james.backends.postgres.utils.PostgresExecutor;
+import org.apache.james.backends.postgres.utils.SimpleJamesPostgresConnectionFactory;
+import org.apache.james.core.Username;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MailboxSessionUtil;
+import org.apache.james.mailbox.exception.SubscriptionException;
+import org.apache.james.mailbox.store.user.SubscriptionMapper;
+import org.apache.james.mailbox.store.user.SubscriptionMapperFactory;
+import org.apache.james.mailbox.store.user.SubscriptionMapperTest;
+import org.apache.james.mailbox.store.user.model.Subscription;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class PostgresSubscriptionMapperRowLevelSecurityTest {
+    @RegisterExtension
+    static PostgresExtension postgresExtension = new PostgresExtension(PostgresSubscriptionModule.MODULE);
+
+    private SubscriptionMapperFactory subscriptionMapperFactory;
+
+    @BeforeEach
+    public void setUp() {
+        subscriptionMapperFactory = session -> new PostgresSubscriptionMapper(new PostgresSubscriptionDAO(new PostgresExecutor(
+            new SimpleJamesPostgresConnectionFactory(postgresExtension.getConnectionFactory())
+                .getConnection(session.getUser().getDomainPart()))));
+    }
+
+    @Test
+    void subscriptionsCanBeAccessedAtTheDataLevelByMembersOfTheSameDomain() throws Exception {
+        Username username = Username.of("bob@domain1");
+        Username username2 = Username.of("alice@domain1");
+        MailboxSession session = MailboxSessionUtil.create(username);
+        MailboxSession session2 = MailboxSessionUtil.create(username2);
+
+        Subscription subscription = new Subscription(username, "mailbox1");
+        subscriptionMapperFactory.getSubscriptionMapper(session)
+            .save(subscription);
+
+        assertThat(subscriptionMapperFactory.getSubscriptionMapper(session2)
+            .findSubscriptionsForUser(username))
+            .containsOnly(subscription);
+    }
+
+    @Disabled("Row level security for subscriptions is not implemented correctly")
```

Comment:
The database issue, the database user default via test container looks like "super" user, 

I tested with another user
```
CREATE USER rlsuser WITH PASSWORD 'secret';
CREATE DATABASE rlsdb;
grant all privileges on database rlsdb to rlsuser;
```
And downgrade postgresql 16 -> 11.1, Then this test passed

Reply:
> And downgrade postgresql 16 -> 11.1, Then this test passed

Forbidden